ambry-replication/src/main/java/com/github/ambry/replication/ReplicationMetrics.java

```diff
@@ -434,6 +437,20 @@ public void addLagMetricForPartition(PartitionId partitionId) {
     }
   }
 
+  /**
+   * Add catchup point metric(local from cloud) for given partitionId.
+   * @param partitionId partition to add metric for.
+   */
+  public void addCatchUpPointMetricForPartition(PartitionId partitionId) {
+    if (!cloudReplicaCatchUpPoint.containsKey(partitionId)) {
+      cloudReplicaCatchUpPoint.put(partitionId, 0L);
+      // Set up metrics if and only if no mapping for this partition before.
+      Gauge<Long> catchUpPoint = () -> cloudReplicaCatchUpPoint.get(partitionId);
+      registry.register(MetricRegistry.name(ReplicaThread.class,
+          "Partition-" + partitionId.toPathString() + "-catchupPointFromCloud"), catchUpPoint);
```

Comment:
minor: I know it wasn't done for the other similar metrics, but since this name format is used in two methods you could probably make a helper method to generate the name
```java
String getCatchupPointFromCloudMetricName(PartitionId partitionId) { 
  return MetricRegistry.name(ReplicaThread.class, "Partition-" + partitionId.toPathString() + "-catchupPointFromCloud");
}
```

Reply:
done.